### DISCUSSION TITLE
Args array is empty in GraphQLType fields() function

### DISCUSSION THREAD

**Discussion by kardashflaxen** [rebing/graphql-laravel#786] [General]
Hello!
I have a problem with empty args array  into GraphQLType fields() function, although it's correct in query resolve().
 
```php
 public function fields(): array
    {
        $fields = parent::fields();

        $fields['name'] = [
            'type' => Type::string(),
        ];

        $fields['checks'] = [
            'type' => Type::listOf(CheckType::type()),
            'is_relation' => true,
        ];

        $fields['user_check'] = [
            'type' => CheckType::type(),
            'resolve' => function (Tariff $tariff, array $args) {
            dd($args);
                return $tariff->checks()->find($args['check_id']);
            }
        ];
}
```

In this case array `$args` is empty. Please, tell me what is wrong with it. 
Thank you!

**Comment by mfn** [comment]
You need to show your actual query.

I suspect you are *not* passing an argument to your `user_check` field, i.e. to receive `$args` you need to have the like this:
```graphql
  …
  yourType {
    user_check(check:_id: …
  }
```
But this also requires defining the `'args'` on that field, e.g.
```php
        $fields['user_check'] = [
            'type' => CheckType::type(),
            'args' => [
                'check_id' => [
                    'type' => …
                ],
            ],
            'resolve' => function (Tariff $tariff, array $args) {
            dd($args);
                return $tariff->checks()->find($args['check_id']);
            }
        ];
```

In the (distant) past, this library _used_ to populate the query/mutation args into the field args, but that was just a plain bug and that functionality was removed.

**Comment by kardashflaxen** [comment]
Thank you for your answer, it is help me. 
My mutation 
`mutation {
  tariffUpdateMutation (id: 10, check_id: 2, user_comment: "Comment") {
    id
    name
    opening_date
    user_check (check_id: 2){
      type
      name
      active
      status
    }
  }
}`

In my TariffType
`$fields['user_check'] = [
            'type' => CheckType::type(),
            'args' => [
                'check_id' => [
                    'type' => Type::int(),
                ],
            ],
            'resolve' => function (Tariff $tariff, array $args) {
                return $tariff->checks()->find($args['check_id']);
            }
        ];`

And it is work! 
Thank you for help!
